spec/dor/services/client/marcxml_spec.rb

```diff
@@ -47,4 +47,86 @@
       end
     end
   end
+
+  describe '#marcxml' do
+    let(:catkey) { '12345678' }
+
+    let(:status) { 200 }
+
+    let(:body) do
+      <<-EOXML
+      <?xml version="1.0"?>
+        <record xmlns="http://www.loc.gov/MARC21/slim">
+        <leader>     cam a22     Ia 4500</leader>
+        <controlfield tag="005">20130318221743.0</controlfield>
+        <controlfield tag="008">921116s1992    moua     b    000 0 eng d</controlfield>
+        <datafield ind1=" " ind2=" " tag="035">
+          <subfield code="a">180345</subfield>
+            </datafield>
+        <datafield ind1=" " ind2=" " tag="040">
+          <subfield code="a">DBI</subfield>
+            <subfield code="c">DBI</subfield>
+          <subfield code="d">OCL</subfield>
+            <subfield code="d">OCLCQ</subfield>
+        </datafield>
+      </record>
+      EOXML
+    end
+
+    before do
+      stub_request(:get, 'https://dor-services.example.com/v1/catalog/marcxml')
+        .with(
+          query: { 'barcode' => barcode }
+        )
+        .to_return(status: status, body: body)
+      stub_request(:get, 'https://dor-services.example.com/v1/catalog/marcxml')
+        .with(
+          query: { 'catkey' => catkey }
+        )
+        .to_return(status: status, body: body)
+    end
+
+    context 'when API request succeeds with barcode and there is a body' do
+      it 'returns the MARCXML' do
+        expect(client.marcxml(barcode: barcode)).to be_a(Nokogiri::XML::Document)
+      end
+    end
+
+    context 'when API request succeeds with catkey and there is a body' do
+      it 'returns the MARCXML' do
+        expect(client.marcxml(barcode: barcode)).to be_a(Nokogiri::XML::Document)
```

Comment:
I think `barcode: barcode` should be `catkey: catkey`

Reply:
Fixed.